src/upkeep.rs

```diff
@@ -300,22 +301,28 @@ pub async fn do_upkeep(
                     let topic = runtime_config.demoted_topic.clone().unwrap_or(config.kafka_long_topic.clone());
 
                     async move {
-                        if inflight.status == InflightActivationStatus::Complete {
-                            return Ok(inflight.id);
-                        }
                         metrics::counter!("upkeep.forward_demoted_namespace", "namespace" => inflight.namespace, "taskname" => inflight.taskname).increment(1);
+
+                        let modified_activation = match modify_activation_namespace(&inflight.activation) {
```

Comment:
This does not mutate `inflight` in place right? It creates a new activation with the changed namespace instead. Is it correct ?  (as `inflight` is used later in this method, mutating it in place would be dangerous)